spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -584,70 +595,91 @@ def __relative_ROIvol_in_mask(self, im_mask_data, im_atlas_roi_data, p_lst, im_t
+        # Loop over slices or vertebral levels
+        for row, indices_to_keep in sct_progress_bar(self.rows.items(), unit=self.row_name,
+                                                     desc="  Computing lesion distribution (volume values)"):
+            if np.count_nonzero(im_vert_and_lesion[indices_to_keep]):  # if there is lesion in this vertebral level
+                idx = self.distrib_matrix_dct[sheet_name][self.distrib_matrix_dct[sheet_name].row == str(row)].index
                 for tract_id in atlas_data:  # Loop over PAM50 tracts
-                    res_lst = self.__relative_ROIvol_in_mask(im_mask_data=np.copy(im_lesion),
-                                                             im_atlas_roi_data=np.copy(atlas_data[tract_id]),
+                    res_lst = self.__relative_ROIvol_in_mask(im_mask_data=im_lesion,
+                                                             im_atlas_roi_data=atlas_data[tract_id],
                                                              p_lst=p_lst,
-                                                             im_template_vert_data=np.copy(im_vert_cur),
-                                                             vert_level=vert)
+                                                             indices_to_keep=indices_to_keep)
                     self.distrib_matrix_dct[sheet_name].loc[idx, 'PAM50_' + str(tract_id).zfill(2)] = res_lst[0]
                     vol_mask_tot += res_lst[0]
 
         # convert the volume values in distrib_matrix_dct to percentage values
-        for vert in self.vert_lst:
-            idx = self.distrib_matrix_dct[sheet_name][self.distrib_matrix_dct[sheet_name].vert == str(vert)].index
+        for row in sct_progress_bar(self.rows.keys(), unit=self.row_name,
+                                    desc="  Converting volume values into percentage values"):
+            idx = self.distrib_matrix_dct[sheet_name][self.distrib_matrix_dct[sheet_name].row == str(row)].index
             for tract_id in atlas_data:
                 val = self.distrib_matrix_dct[sheet_name].loc[idx, 'PAM50_' + str(tract_id).zfill(2)].values[0]
                 self.distrib_matrix_dct[sheet_name].loc[idx, 'PAM50_' + str(tract_id).zfill(2)] = val * 100.0 / vol_mask_tot
 
-    def __regroup_per_tracts(self, vol_dct, tract_limit):
-        res_mask = [vol_dct[t][0] for t in vol_dct if t >= tract_limit[0] and t <= tract_limit[1]]
-        res_tot = [vol_dct[t][1] for t in vol_dct if t >= tract_limit[0] and t <= tract_limit[1]]
+        # Add the total column
+        self.distrib_matrix_dct[sheet_name]['total % (all tracts)'] = \
+            self.distrib_matrix_dct[sheet_name].sum(numeric_only=True, axis=1)
+
+        # Add additional columns for the "CombinedLabels" defined by info_label.txt
+        for label_name, sublabels in self.atlas_combinedlabels.items():
+            column_names_to_sum = [f"PAM50_{str(subid).zfill(2)}" for subid in sublabels]
```

Comment:
This format spec should be equivalent for numbers:
```suggestion
            column_names_to_sum = [f"PAM50_{subid:02}" for subid in sublabels]
```